### PR TITLE
Typing generator

### PR DESCRIPTION
**Pull request by camnwalter**
Still in progress
Adds:
- instance methods for kotlin objects
- vals/vars in constructors now have appropriate getters/setters
- private/internal getters/setters are now hidden
- static classes now have appropriate fields in their enclosing classes

### CODE REVIEW

File: typing-generator/src/main/kotlin/com/chattriggers/ctjs/typing/Provider.kt [hunk 1]
```diff
@@ -234,6 +244,10 @@ class Processor(environment: SymbolProcessorEnvironment) : SymbolProcessor {
         })
 
         indented {
+            if (clazz.classKind == ClassKind.OBJECT) {
+                staticProperties.forEach { buildProperty(it, resolver) }
+                staticFunctions.forEach { buildFunction(it, resolver, omitName = false) }
+            }
```
Comment (mattco98) [inline typing-generator/src/main/kotlin/com/chattriggers/ctjs/typing/Provider.kt:250]: So if the class is `object`, then the static properties will be output for both the instance type and the static type? That makes sense to me, I just want to make sure I'm reading it right

File: build.gradle.kts [hunk 2]
```diff
@@ -66,7 +66,8 @@ java {
 }
 
 apiValidation {
-    ignoredPackages.add("com.chattriggers.ctjs.internal")
+    ignoredProjects += "typing-generator"
+    ignoredPackages += "com.chattriggers.ctjs.internal"
```
Comment (mattco98) [inline build.gradle.kts:70]: Does this remove our dependency on EssentialGG/essential-gradle-toolkit#15?

File: typing-generator/src/main/kotlin/com/chattriggers/ctjs/typing/Provider.kt [hunk 3]
```diff
@@ -173,9 +173,15 @@ class Processor(environment: SymbolProcessorEnvironment) : SymbolProcessor {
...
-        // Output static object first, if necessary
-        if (staticProperties.isNotEmpty() || staticFunctions.isNotEmpty() || isEnum) {
-            appendLine("const $name: {")
+        val nestedClasses = clazz.declarations.filterIsInstance<KSClassDeclaration>().filter {
+            it.isPublic()
+        }.filter {
+            it.classKind == ClassKind.ENUM_CLASS || it.classKind == ClassKind.CLASS
+        }.toList()
+
+        // Output static object first always, could be empty
```
[the remark this answers is not in the document]
Comment (camnwalter) [reply typing-generator/src/main/kotlin/com/chattriggers/ctjs/typing/Provider.kt:182]: Oops that was a remnant of something I ended up removing.

File: typing-generator/src/main/kotlin/com/chattriggers/ctjs/typing/Provider.kt (hunk 1 above)
Comment (camnwalter) [reply typing-generator/src/main/kotlin/com/chattriggers/ctjs/typing/Provider.kt:250]: Yes exactly

File: build.gradle.kts (hunk 2 above)
Comment (camnwalter) [reply build.gradle.kts:70]: We already migrated away from egt in a previous PR.

Comment (mattco98) [reply build.gradle.kts:70]: so true

**Pull request merged**